packages/assets-controllers/src/TokenListController.ts

```diff
@@ -286,80 +286,67 @@ export class TokenListController extends StaticIntervalPollingController<TokenLi
   async fetchTokenList(chainId: Hex): Promise<void> {
     const releaseLock = await this.mutex.acquire();
     try {
-      const { tokensChainsCache } = this.state;
-      let tokenList: TokenListMap = {};
-      // Attempt to fetch cached tokens
-      const cachedTokens = await safelyExecute(() =>
-        this.#fetchFromCache(chainId),
+      if (this.isCacheValid(chainId)) {
+        return;
+      }
```

Comment:
This line does the heavy lifting.
- Before: we never early returned and always performed state updates
- Now: we early return to prevent additional state updates.

Reply:
The rest of the changes below are some cleanups to help reduce state updates and inform immer on state that actually changed (e.g. no spreading to create more new objects!)